scripts/build.ps1

```diff
@@ -95,7 +98,10 @@ param (
     [switch]$DynamicCRT = $false,
 
     [Parameter(Mandatory = $false)]
-    [switch]$PGO = $false
+    [switch]$PGO = $false,
+
+    [Parameter(Mandatory = $false)]
+    [string]$Generator = "Linux Makefiles"
```

Comment:
Instead of defaulting for Linux, you should default to empty, and below set platform specific generators if it's empty.